Source/Core/DiscIO/DiscExtractor.cpp

```diff
@@ -212,7 +227,17 @@ bool ExportHeader(const Volume& volume, const Partition& partition,
   if (!IsDisc(volume.GetVolumeType()))
     return false;
 
-  return ExportData(volume, partition, DISCHEADER_ADDRESS, DISCHEADER_SIZE, export_filename);
+  if (!ExportData(volume, partition, DISCHEADER_ADDRESS, DISCHEADER_SIZE, export_filename))
+    return false;
+
+  if (volume.IsNKit())  // Erase NKitv1 data
+  {
+    static constexpr std::array<std::byte, 0x1C> null_bytes = {std::byte{0}};
+    File::IOFile f(export_filename, "r+b");
+    f.Seek(0x200, File::SeekOrigin::Begin);
+    f.WriteBytes(null_bytes.data(), null_bytes.size());
```

Comment:
> I had considered making the DiscIO::ExportData function return the File::IOFile class it constructs, allowing the caller to either immediately convert it to a bool, or carry on using it. This would remove the need to open the file a second time in ExportWiiUnencryptedHeader and ExportHeader for these minor edits, though due to the File::IOFile operator bool overload being explicit, I thought it looked kind of ugly. Would this still be preferred over opening the file a second time?

That's not a bad idea. But I don't understand why you would need to do anything ugly. Is it not possible to just use `!`, like here? https://github.com/dolphin-emu/dolphin/blob/2049ff96c5c140f872510c756327ace9e4e8ab42/Source/Core/DiscIO/DiscExtractor.cpp#L57

Reply:
The issue is every other function which uses the ExportData function without inverting the result either needs to do the double NOT idiom (I am not really fond of this), or cast to bool (which is noisy when very few examples need this feature).  Ultimately, the best solution is the one I didn't come up with yesterday: Make a seperate function which returns the File::IOFile rather than a bool, so anything which does not care can continue using the original.